deepmd/entrypoints/test.py

```diff
@@ -531,8 +543,33 @@ def test_ener(
             "pred_vyy pred_vyz pred_vzx pred_vzy pred_vzz",
             append=append_detail,
         )
+        if dp.has_hessian:
+            _n_frames_, _n_hessian_ = test_data["hessian"][:numb_test].shape
+            _n_atoms_ = np.int32(np.sqrt(_n_hessian_) / 3)  # n_hessian = 3na*3na
+            triu_indices = np.triu_indices(
+                _n_atoms_ * 3
+            )  # upper triangle hessian indices
+            data_h_triu = test_data["hessian"][:numb_test][
+                :, triu_indices[0] * _n_atoms_ * 3 + triu_indices[1]
+            ].reshape(-1, 1)
+            pred_h_triu = hessian[
+                :, triu_indices[0] * _n_atoms_ * 3 + triu_indices[1]
+            ].reshape(-1, 1)
+            h = np.concatenate(
+                (
+                    data_h_triu,
+                    pred_h_triu,
+                ),
+                axis=1,
+            )
+            save_txt_file(
+                detail_path.with_suffix(".h.out"),
+                h,
+                header=f"{system}: data_h pred_h",
```

Comment:
Since the shape of Hessian matrices vary from frames with different numbers of atoms,  the Hessian matrices can only be reshaped to `(-1, )` before being printed to the `detail.h.out` file. Therefore, the header needs to discard the information of the order of elements and retain only two columns, namely `data_h` and `pred_h`. 
For example, testing two systems with numbers of atoms of 15 and 18 respectively, using `--numb-test 1`, the shape of `detail.h.out` is `(4941, 2)`, where 4941=(15\*3)^2+(18\*3)^2.